examples/azure/client_factory.md

```diff
@@ -0,0 +1,112 @@
+
+This Environment structure has the following fields configured for each cloud environment:
+
+```golang
+ManagementPortalURL          
+PublishSettingsURL           
+ServiceManagementEndpoint    
+ResourceManagerEndpoint      
+ActiveDirectoryEndpoint      
+GalleryEndpoint              
+KeyVaultEndpoint             
+GraphEndpoint                
+ServiceBusEndpoint           
+BatchManagementEndpoint      
+StorageEndpointSuffix        
+SQLDatabaseDNSSuffix         
+TrafficManagerDNSSuffix      
+KeyVaultDNSSuffix            
+ServiceBusEndpointSuffix     
+ServiceManagementVMDNSSuffix 
+ResourceManagerVMDNSSuffix   
+ContainerRegistryDNSSuffix   
+CosmosDBDNSSuffix            
+TokenAudience                
+APIManagementHostNameSuffix  
+SynapseEndpointSuffix        
+ResourceIdentifiers          
+```
+
+Using these URI's, `client_factory` is able to instantiate and return properly configured SDK clients without module developers having to duplicate this code.
+
+## Configuration and Defaults
+
+To configure different cloud environments, we will use the same `AZURE_ENVIRONMENT` environment variable that the Go SDK uses. This can be set to one of the following values:
+
+|Value                      |Cloud Environment  |
+|---------------------------|-------------------|
+|"AzureChinaCloud"          |ChinaCloud         |
+|"AzureGermanCloud"         |GermanCloud        |
+|"AzurePublicCloud"         |PublicCloud        |
+|"AzureUSGovernmentCloud"   |USGovernmentCloud  |
+|"AzureStackCloud"          |Azure stack        |
+
+Note that when using the "AzureStackCloud" setting, you must also set the `AZURE_ENVIRONMENT_FILEPATH` variable to point to a JSON file containing your Azure Stack URI deatils.
+
+>NOTE: The default behavior of the `client_factory` is to use the AzurePublicCloud environment. This requires no work from the developer to configure, and ensures consistent behavior with the current SDK code.       
+
+## Usage Patterns
+
+Modules authors will interact with the `client_factory` through the `CreateXXXXClientE` methods on the `azure` package as shown in the following example:
+
+```golang
+    // Create a new client instance
+	client, err := CreateVirtualMachinesClientE(VirtualMachinesClientType, subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+```
+
+The `client_factory` currently exposes the following methods for creating clients. These will be built out more as more modules are developed and integrated:
+
+* `CreateSubscriptionsClientE() (subscriptions.Client, error)`: Returns a virtual machines client instance configured with the correct BaseURI depending on the Azure environment that is currently setup (or "Public", if none is setup).
```

Comment:
I guess the question I was trying to answer was "what's available?". But if the answer to that is "go read the source", then yes these can be removed.

Reply:
Each method should have comments since they are public, I think a short list here of ones that are best practices (IE, the ones you have in initial PR is fine, I would just update wording to say these show the pattern we want you dev to follow.)